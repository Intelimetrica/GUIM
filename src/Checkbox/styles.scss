@import "../styles/colors.scss";
@import "../styles/themes.scss";

.GUIMCheckbox {
  -webkit-appearance: none;
  background-color: var(--white);
  border: 2px solid var(--main-gray);
  border-radius: 3px;
  display: inline-block;
  position: relative;
  outline: none;
  width: 16px;
  height: 16px;
  transition: all 0.5s ease;
  cursor: pointer;
  &:checked {
    &::before {
      -webkit-transform: rotate(45deg);
      transform: rotate(45deg);
      content: "";
      position: absolute;
      top: 0px;
      left: 5px;
      display: table;
      width: 2px;
      height: 10px;
      border: 2px solid #fff;
      border-top-width: 0;
      border-left-width: 0;
    }
  }
  &:disabled {
    cursor: default;
    opacity: 0.4;
  }
}
.GUIMSwitch {
  display: flex;
  width: 50px;
  height: 22px;
  &.Disabled > * {
    opacity: 0.4;
    cursor: not-allowed !important;
  }
}

.GUIMSwitch label {
  cursor: pointer;
  text-indent: -9999px;
  width: 40px;
  height: 22px;
  background: #bbbbbb;
  display: block;
  border-radius: 90px;
  position: relative;
}
.GUIMSwitch input[type="checkbox"] {
  height: 0;
  width: 0;
  visibility: hidden;
}
.GUIMSwitch label:after {
  content: "";
  position: absolute;
  top: 1px;
  left: 1px;
  width: 18px;
  height: 18px;
  background: #fff;
  border-radius: 90px;
  transition: 0.1s;
}
.GUIMSwitch input:not(:checked) + label:after {
  top: 2px;
  left: 2px;
}
.GUIMSwitch.Checked label {
  height: 20px;
  width: 38px;
}
.GUIMSwitch input:checked + label:after {
  left: calc(100% - 1px);
  transform: translateX(-100%);
}
.GUIMSwitch label:active:after {
  width: 18px;
}

.GUIMCheckboxBlue:checked {
  @apply --blue-theme;
}
.GUIMCheckboxGray:checked {
  @apply --gray-theme;
}
.GUIMCheckboxGreen:checked {
  @apply --green-theme;
}
.GUIMCheckboxOrange:checked {
  @apply --orange-theme;
}
.GUIMSwitchBlue:checked + label {
  @apply --blue-theme;
}
.GUIMSwitchGray:checked + label {
  @apply --gray-theme;
}
.GUIMSwitchGreen:checked + label {
  @apply --green-theme;
}
.GUIMSwitchOrange:checked + label {
  @apply --orange-theme;
}
