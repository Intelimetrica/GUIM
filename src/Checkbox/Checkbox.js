import React from "react";
import "./styles.scss";

const Checkbox = props => {
  let id = null;
  if (props.id != "notIdSet"){
    id = props.id;
  } else {
    id = Math.random();
  }
  const check = (
    <input
      id={`switch-${id}`}
      className={props.isSwitch ?`Switch ${themesSwitch[props.color] || themesSwitch["blue"]}` : `GUIMCheckbox ${props.className} ${themes[props.color] || themes["blue"]}`}
      style={props.styles}
      type="checkbox"
      name={props.name}
      onChange={() => props.disabled ? null : props.onChange()}
      checked={props.checked}
      disabled={props.disabled}
    />
  );
  return props.isSwitch ? (
    <div className={`GUIMSwitch ${props.checked ? 'Checked' : ''}`}>
      {check}
      <label htmlFor={`switch-${id}`}>.</label>
    </div>
    ) : (
      check
    )
};

Checkbox.defaultProps = {
  name: "checkbox",
  onChange: () => console.log("Checkbox onChange()"),
  checked: false,
  styles: {},
  className: "",
  color: "blue",
  guimInput: "checkbox",
  disabled: false,
  isSwitch: false,
  id: 'notidset'
};

const themes = {
  blue: "GUIMCheckboxBlue",
  gray: "GUIMCheckboxGray",
  green: "GUIMCheckboxGreen",
  orange: "GUIMCheckboxOrange"
};

const themesSwitch = {
  blue: "GUIMSwitchBlue",
  gray: "GUIMSwitchGray",
  green: "GUIMSwitchGreen",
  orange: "GUIMSwitchOrange"
};

export default Checkbox;
