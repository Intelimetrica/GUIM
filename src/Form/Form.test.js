import React from 'react';
import Form from './Form';
import Checkbox from '../Checkbox';
import Picker from '../Picker';
import Slider from '../Slider';

import renderer from 'react-test-renderer';

import { mount, configure } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';
import Adapter from 'enzyme-adapter-react-16';

expect.addSnapshotSerializer(enzymeSerializer);
configure({ adapter: new Adapter() });

describe('<Form />', () => {
  let active = [1];
  let checked = false;
  let component;
  let form;
  let selected_range = { min: 2, max: 4 };

  let div = document.createElement("div");
  div.setAttribute("id", "slider-tests");
  document.body.appendChild(div);

  const reRender = () => {
    component = renderer.create(
      <Form ref={ref => (form = ref)} >
        <Checkbox
          name="checkbox"
          checked={checked}
          onChange={() => checked = !checked}
          id='Checkbox-1' />
        <Picker
          onChange={(label, value) => active = [value]}
          options={[
            {label: "p1", value: 1},
            {label: "p2", value: 2}
          ]}
          active={active}
        />
        <Slider
          id={"slider-tests"}
          range={{min: 1, max: 5}}
          steps={[1,2,3,4,5]}
          selected_range={selected_range}
          onChange={(new_range) => {selected_range = {...selected_range, new_range}}}
        />
      </Form>)
  };

  it('match dom snapshot', () => {
    component = mount(
      <Form ref={ref => (form = ref)} >
        <Checkbox
          name="checkbox"
          checked={checked}
          onChange={() => checked = !checked} 
          id='Checkbox-2'/>
        <Picker
          onChange={(label, value) => active = [value]}
          options={[
            {label: "p1", value: 1},
            {label: "p2", value: 2}
          ]}
          active={active}
        />
      </Form>);

    expect(component).toMatchSnapshot();
  });

  it('returns children data', () => {
    reRender();
    // Grab data for the first time
    expect(form.grabFormData()).toMatchSnapshot();
  });

  it('returns children data updated after click', () => {
    // click on checkbox to change data
    let tree = component.toJSON();
    tree.children[0].props.onChange();

    reRender();
    expect(form.grabFormData()).toMatchSnapshot();
  });
});
