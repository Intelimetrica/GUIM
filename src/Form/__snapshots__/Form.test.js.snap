// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> match dom snapshot 1`] = `
<Form
  className="GUIMForm"
  guimInput="form"
  name="form"
  styles={Object {}}
>
  <div
    className="GUIMForm"
    name="form"
    style={Object {}}
  >
    <Checkbox
      checked={false}
      className=""
      color="blue"
      disabled={false}
      guimInput="checkbox"
      id="Checkbox-2"
      isSwitch={false}
      name="checkbox"
      onChange={[Function]}
      styles={Object {}}
    >
      <input
        checked={false}
        className="GUIMCheckbox  GUIMCheckboxBlue"
        disabled={false}
        id="switch-Checkbox-2"
        name="checkbox"
        onChange={[Function]}
        style={Object {}}
        type="checkbox"
      />
    </Checkbox>
    <Picker
      active={
        Array [
          1,
        ]
      }
      className=""
      color="blue"
      guimInput="picker"
      name="picker"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "p1",
            "value": 1,
          },
          Object {
            "label": "p2",
            "value": 2,
          },
        ]
      }
    >
      <div
        className="GUIMPicker  GUIMPickerBlue"
        name="picker"
      >
        <Pill
          active={true}
          key="1"
          label="p1"
          select={[Function]}
          value={1}
        >
          <a
            className="active"
            key="1"
            onClick={[Function]}
          >
            p1
          </a>
        </Pill>
        <Pill
          active={false}
          key="2"
          label="p2"
          select={[Function]}
          value={2}
        >
          <a
            className=""
            key="2"
            onClick={[Function]}
          >
            p2
          </a>
        </Pill>
      </div>
    </Picker>
  </div>
</Form>
`;

exports[`<Form /> returns children data 1`] = `
Object {
  "checkbox": false,
  "picker": Array [
    1,
  ],
  "slider": Object {
    "max": 4,
    "min": 2,
  },
}
`;

exports[`<Form /> returns children data updated after click 1`] = `
Object {
  "checkbox": true,
  "picker": Array [
    1,
  ],
  "slider": Object {
    "max": 4,
    "min": 2,
  },
}
`;
